apps/framework-cli/src/cli.rs

```diff
@@ -625,7 +626,11 @@ pub async fn cli_run() {
 
     let cli = Cli::parse();
 
-    match top_command_handler(config, &cli.command).await {
+    let (metrics, rx) = Metrics::new();
+    let arc_metrics = Arc::new(metrics);
+    arc_metrics.clone().start_listening_to_metrics(rx).await;
```

Comment:
Do you need the `clone` here? `start_listening_to_metrics` only needs a reference so it should not be taking ownership